charts/agent/templates/_helpers.tpl

```diff
@@ -537,25 +537,83 @@ true
 {{- end }}
 {{- end }}
 
+{{/* Check if semver. The regex is from the code of the library Helm uses for semver. */}}
+{{- define "agent.isSemVer" -}}
+    {{- if regexMatch "^v?([0-9]+)(\\.[0-9]+)?(\\.[0-9]+)?(-([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?(\\+([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?$" . }}
+        true
+    {{- end -}}
+{{- end -}}
+
 {{/* Return the name of the local forwarder configmap */}}
 {{- define "agent.localForwarderConfigMapName" }}
 {{- include "agent.configmapName" . | trunc 46 | trimSuffix "-" | printf "%s-local-forwarder" }}
 {{- end }}
 
 {{- define "agent.enableHttpProbes" }}
 {{- if not (include "agent.gke.autopilot" .) }}
-{{- if regexMatch "^v?([0-9]+)(\\.[0-9]+)?(\\.[0-9]+)?(-([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?(\\+([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?$" .Values.image.tag }}
-{{- if semverCompare ">= 12.18.0-0" .Values.image.tag }}
+{{- if and (include "agent.isSemVer" .Values.image.tag) (semverCompare ">= 12.18.0-0" .Values.image.tag) }}
 {{- printf "true" -}}
 {{- end }}
 {{- end }}
 {{- end }}
-{{- end }}
 
 {{- define "agent.enableFalcoBaselineSecureLight" }}
-{{- if regexMatch "^v?([0-9]+)(\\.[0-9]+)?(\\.[0-9]+)?(-([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?(\\+([0-9A-Za-z\\-]+(\\.[0-9A-Za-z\\-]+)*))?$" .Values.image.tag }}
-{{- if semverCompare ">= 12.19.0-0" .Values.image.tag }}
+{{- if and (include "agent.isSemVer" .Values.image.tag) (semverCompare ">= 12.19.0-0" .Values.image.tag) }}
 {{- printf "true" -}}
 {{- end }}
 {{- end }}
+
+{{- define "agent.annotationsSection" }}
+  {{- if (include "agent.gke.autopilot" .) }}
+    {{- printf "annotations:" }}
+    {{- printf "autopilot.gke.io/no-connect: \"true\"" | nindent 2 }}
+  {{- else if or (.Values.daemonset.annotations) (eq "false" (include "agent.privileged" .)) }}
+    {{- printf "annotations:" }}
+    {{- if (eq "false" (include "agent.privileged" .)) }}
+      {{- printf "container.apparmor.security.beta.kubernetes.io/sysdig: unconfined" | nindent 2 }}
+    {{- end }}
+    {{- if .Values.daemonset.annotations }}
+      {{- toYaml .Values.daemonset.annotations | nindent 2 }}
+    {{- end }}
+  {{- end }}
+{{- end }}
+
+{{/*
+  - .Values.privileged is true: no problem
+  - .Values.privileged is false:
+    - eBPF disabled: fail
+    - eBPF enabled:
+      - image tag >= 13.3.0: no problem
+      - image tag not semver and has 'dev' in it: go on at user's risk
+*/}}
+{{- define "agent.privileged" }}
+  {{- if or .Values.privileged (include "agent.gke.autopilot" .) }}
+    {{- /* OK */ -}}
+    {{- print "true" }}
+  {{- else }}
+    {{- $errMsg := "Disabling 'privileged' flag requires eBPF and Sysdig Agent 13.3.0 or newer" }}
+    {{- /* eBPF is mandatory */ -}}
+    {{- if not (eq "true" (include "agent.ebpfEnabled" .)) }}
```

Comment:
I think we should check for Universal eBPF instead of all the eBPF flavour here...

Reply:
Why is that so? `legacy_ebpf` can run with specific capabilities